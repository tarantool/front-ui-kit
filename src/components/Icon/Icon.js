// @flow
import * as React from 'react'
import { css, cx } from '@emotion/css';
import { rgba } from 'emotion-rgba';
import { colors, iconSize } from '../../variables';
import { SVGImage } from '../SVGImage';

const styles = {
  icon: css`
    flex-shrink: 0;
    vertical-align: middle;
    width: ${iconSize};
    height: ${iconSize};
  `,
  state: css`
    fill: red;

    &:hover {
      fill: greenyellow;
    }

    &.active {
      fill: blue;
    }
  `,
  stroke: css`
    stroke: red;

    &:hover {
      fill: greenyellow;
    }

    &.active {
      fill: blue;
    }
  `,
  clickable: css`
    cursor: pointer;
  `,
  active: css``,
  button: css`
    display: block;
    padding: 0;
    border: none;
    outline: none;
    background: transparent;

    &:focus::before {
      content: '';
      position: absolute;
      top: -2px;
      left: -2px;
      right: -2px;
      bottom: -2px;
      border: solid 1px ${rgba(colors.intentPrimary, 0.55)};
      border-radius: 3px;
    }
  `
};

type IconProps = {
  active?: boolean; // Выбраное состояние
  className?: string;
  glyph: SVGGlyph,
  hasState?: boolean, // Включение состояния: Normal, Hover, Active
  onClick?: (evt: MouseEvent) => void,
  onMouseEnter?: () => void,
  onMouseLeave?: () => void,
  stroke?: boolean // Задавать stroke вместо fill
};

export type GenericIconProps = {
  className?: string,
  onClick?: (e: MouseEvent) => void
};


export const Icon = React.memo<IconProps>(({
  active,
  className,
  glyph,
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  hasState, // TODO: 'hasState' is defined but never used.
  onMouseLeave,
  onMouseEnter,
  onClick,
  stroke
}: IconProps) => (
  <SVGImage
    glyph={glyph}
    onClick={onClick}
    onMouseLeave={onMouseLeave}
    onMouseEnter={onMouseEnter}
    className={cx(
      styles.icon,
      {
        [styles.stroke]: stroke,
        [styles.clickable]: !!onClick,
        [styles.active]: active
      },
      className
    )}
  />
));
