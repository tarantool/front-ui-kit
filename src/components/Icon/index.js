// @flow
import * as React from 'react'
import { css, cx } from 'emotion';
import { iconSize } from '../../variables';

const styles = {
  icon: css`
    vertical-align: middle;
    width: ${iconSize};
    height: ${iconSize};
  `,

  state: css`
    fill: red;

    &:hover {
      fill: greenyellow;
    }

    &.active {
      fill: blue;
    }
  `,

  stroke: css`
    stroke: red;

    &:hover {
      fill: greenyellow;
    }

    &.active {
      fill: blue;
    }
  `,

  clicked: css`
    cursor: pointer;
  `,

  active: css``
};

type Glyph = {
  content: string,
  id: string,
  node: any,
  viewBox: string,
};

interface IconProps {
  active?: boolean; // Выбраное состояние
  className?: string;
  glyph: Glyph,
  hasState?: boolean, // Включение состояния: Normal, Hover, Active
  onClick?: (evt: MouseEvent) => void,
  onMouseEnter?: () => void,
  onMouseLeave?: () => void,
  stroke?: boolean // Задавать stroke вместо fill
}

const Icon = ({
  active,
  className,
  glyph,
  hasState,
  onMouseLeave,
  onMouseEnter,
  onClick,
  stroke
}: IconProps) => {
  const isInteractive = onClick || onMouseEnter || onMouseLeave;

  const svg = (
    <svg
      viewBox={glyph.viewBox}
      className={cx(
        styles.icon,
        {
          [styles.state]: hasState,
          [styles.stroke]: stroke,
          [styles.clicked]: !!onClick,
          [styles.active]: active
        },
        className,
      )}
    >
      <use xlinkHref={`#${glyph.id}`}/>
    </svg>
  );

  return isInteractive
    ? (
      <button
        className={cx(
          styles.icon,
          {
            [styles.state]: hasState,
            [styles.stroke]: stroke,
            [styles.clicked]: !!onClick,
            [styles.active]: active
          },
          className,
        )}
        onClick={onClick}
        onMouseLeave={onMouseLeave}
        onMouseEnter={onMouseEnter}
        type='button'
      >
        {svg}
      </button>
    )
    : svg;
};

export default Icon;
