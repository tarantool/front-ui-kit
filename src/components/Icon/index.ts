export * from './Icon';
export * from './icons/IconArrow';
export * from './icons/IconCheckbox';
export * from './icons/IconChevron';
export * from './icons/IconSpinner';
