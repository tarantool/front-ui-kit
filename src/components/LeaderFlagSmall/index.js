// @flow
import * as React from 'react';
import { css, cx } from 'emotion';
import glyph from './flagSmall.svg';

const styles = {
  flag: css`
    width: 42px;
    height: 8px;
  `
};

type LeaderFlagSmallProps = {
  className?: string,
};

export const LeaderFlagSmall = ({ className }: LeaderFlagSmallProps) => (
  <svg
    viewBox={glyph.viewBox}
    className={cx(styles.flag, className)}
  >
    <use xlinkHref={`#${glyph.id}`}/>
  </svg>
);
