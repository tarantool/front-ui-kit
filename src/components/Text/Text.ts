import { MouseEvent, ReactNode, createElement, forwardRef } from 'react';
import { cx } from '@emotion/css';

import { textStyles } from './Text.styles';

export { textStyles };

export type TextStyleVariants = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'p' | 'basic' | 'code';

export interface TextProps {
  className?: string;
  htmlFor?: string;
  children?: ReactNode;
  tag?: string;
  href?: string;
  upperCase?: boolean;
  noCase?: boolean;
  title?: string;
  variant?: TextStyleVariants;
  onClick?: (event: MouseEvent<HTMLElement>) => void;
  // colorVariant?: 'inherit' | 'primary' | 'secondary' | 'disabled' | 'white' | 'error',
}

export const Text = forwardRef<HTMLElement, TextProps>(
  ({ className, children, tag, upperCase, noCase, variant = 'basic', ...props }, ref) => {
    const Element = tag || (variant === 'basic' ? 'span' : variant);

    return createElement(
      Element,
      {
        ...props,
        className: cx(
          textStyles[variant],
          {
            [textStyles.upperCase]: upperCase,
            [textStyles.noCase]: noCase,
          },
          className
        ),
        ref,
      },
      children
    );
  }
);
