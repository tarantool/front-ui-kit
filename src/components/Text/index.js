// @flow
import * as React from 'react'
import { css, cx } from 'emotion'

export const textStyles = {
  h1: css`
    margin: 0;
    font-family: 'Open Sans', sans-serif;
    font-size: 24px;
    line-height: 38px;
    font-weight: 600;
    letter-spacing: 0.48px;
    text-transform: uppercase;
    color: #000000;
  `,
  h2: css`
    margin: 0;
    font-family: 'Open Sans', sans-serif;
    font-size: 20px;
    line-height: 28px;
    font-weight: 600;
    letter-spacing: 0.4px;
    text-transform: uppercase;
    color: #000000;
  `,
  h3: css`
    margin: 0;
    font-family: 'Open Sans', sans-serif;
    font-size: 16px;
    line-height: 24px;
    font-weight: 600;
    letter-spacing: 0.32px;
    text-transform: uppercase;
    color: #cf1322;
  `,
  h4: css`
    margin: 0;
    font-family: 'Open Sans', sans-serif;
    font-size: 16px;
    line-height: 24px;
    font-weight: 600;
    letter-spacing: 0.32px;
    color: #000000;
  `,
  h5: css`
    margin: 0;
    font-family: 'Open Sans', sans-serif;
    font-size: 12px;
    line-height: 22px;
    letter-spacing: 0.24px;
    color: #000000;
  `,
  p: css`
    margin: 0;
    font-family: 'Open Sans', sans-serif;
    font-size: 12px;
    line-height: 20px;
    letter-spacing: 0.28px;
    color: #000000;
  `,
  code: css`
    margin: 0;
    font-family: 'Monaco', monospace;
    font-size: 14px;
    line-height: 22px;
    color: #000000;
    b {
      font-weight: 600;
    }
  `,
  basic: css`
    margin: 0;
    font-family: 'Open Sans', sans-serif;
    font-size: 14px;
    line-height: 22px;
    color: #000000;
    b {
      font-weight: 600;
    }
  `,
  upperCase: css`
    text-transform: uppercase;
  `,
  noCase: css`
    text-transform: none;
  `
};

type TextProps = {
  className?: string,
  children?: React.Node,
  tag?: string,
  upperCase?: boolean,
  noCase?: boolean,
  title?: string,
  variant?:
    |'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'p' | 'basic' | 'code',
  // colorVariant?: 'inherit' | 'primary' | 'secondary' | 'disabled' | 'white' | 'error',
};

export const Text = ({
  className,
  children,
  tag,
  upperCase,
  noCase,
  title,
  variant = 'basic'
}:
TextProps) => {
  const Element = tag || (variant === 'basic' ? 'span' : variant);

  return (
    <Element
      className={cx(
        textStyles[variant],
        {
          [textStyles.upperCase]: upperCase,
          [textStyles.noCase]: noCase
        },
        className
      )}
      title={title}
    >
      {children}
    </Element>
  );
};
