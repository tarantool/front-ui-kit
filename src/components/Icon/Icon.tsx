import React, { FC, MouseEvent } from 'react';
import { cx } from '@emotion/css';

import { SVGImage } from '../SVGImage';
import { styles } from './Icon.styled';

type IconProps = {
  active?: boolean; // Выбраное состояние
  className?: string;
  glyph?: SVGGlyphTypes;
  hasState?: boolean; // Включение состояния: Normal, Hover, Active
  onClick?: (evt: MouseEvent) => void;
  onMouseEnter?: () => void;
  onMouseLeave?: () => void;
  stroke?: boolean; // Задавать stroke вместо fill
};

export type GenericIconProps = {
  className?: string;
  onClick?: (e: MouseEvent) => void;
};

export type SVGGlyphTypes = {
  content: string;
  id: string;
  node: SVGSVGElement;
  viewBox: string;
};

export const Icon: FC<IconProps> = React.memo<IconProps>(
  ({
    active,
    className,
    glyph,
    // hasState, // TODO: 'hasState' is defined but never used.
    onMouseLeave,
    onMouseEnter,
    onClick,
    stroke,
  }: IconProps) => (
    <SVGImage
      glyph={glyph}
      onClick={onClick}
      onMouseLeave={onMouseLeave}
      onMouseEnter={onMouseEnter}
      className={cx(
        styles.icon,
        {
          [styles.stroke]: stroke,
          [styles.clickable]: !!onClick,
          [styles.active]: active,
        },
        className
      )}
    />
  )
);
