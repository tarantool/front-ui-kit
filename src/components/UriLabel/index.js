// @flow
import React from 'react';
import { css, cx } from '@emotion/css';

import { colors } from '../../variables';
import { IconBoldArrowRight, IconLink } from '../Icon';
import { Text } from '../Text';

const styles = {
  uriWrap: css`
    display: flex;
    align-items: center;
  `,
  uriIcon: css`
    width: 14px;
    height: 14px;
    margin-right: 4px;
  `,
  uri: css`
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    color: rgba(0, 0, 0, 0.65);
  `,
  hereIcon: css`
    fill: ${colors.intentWarningAccent};
  `,
};

const IconHere = ({ className, ...props }) => (
  <IconBoldArrowRight className={cx(styles.hereIcon, className)} {...props} />
);

type UriLabelProps = {
  className?: string,
  icon?: React.AbstractComponent<any>,
  title?: string,
  uri?: string,
  weAreHere?: boolean,
};

export const UriLabel = ({ className, icon, title, uri, weAreHere }: UriLabelProps) => {
  const Icon = icon || (weAreHere ? IconHere : IconLink);

  return (
    <div className={cx(styles.uriWrap, className)} title={title}>
      <Icon className={styles.uriIcon} />
      <Text className={styles.uri} variant="p" tag="span">
        {uri}
      </Text>
    </div>
  );
};
