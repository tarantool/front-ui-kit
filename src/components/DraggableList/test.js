import React from 'react';
import renderer from 'react-test-renderer';
import { DraggableList, DraggableListItem } from './index';
import { Text } from '../Text';
import { IconBurger } from '../Icon';
import { LeaderFlagSmall } from '../LeaderFlagSmall';
import { UriLabel } from '../UriLabel';

const noop = () => void 0;

it('DraggableList renders correctly', () => {
  const data = Array.apply(null, { length: 10 }).map((_, index) => ({
    key: index,
    uuid: index,
    alias: `alias ${index}`,
    uri: `uri ${index}`
  }));

  const items = data.map((item, index) => (
    <DraggableListItem
      num={index}
      index={index}
      key={item.key}
    >
      <div>
        <Text tag='div'>
          <IconBurger />
          {item.alias || item.uuid}
        </Text>
        <LeaderFlagSmall/>
        <UriLabel
          uri={item.uri}
        />
      </div>
    </DraggableListItem>
  ));

  const tree = renderer.create(
    <DraggableList items={items} onChange={noop} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
