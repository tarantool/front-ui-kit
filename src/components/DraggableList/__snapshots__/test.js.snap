// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraggableList renders correctly 1`] = `
<div
  className=""
>
  <div
    className=""
  >
    <div>
      <div
        className="css-1ggp6x0"
      >
        <svg
          className="css-1f5lesb"
          viewBox="0 0 16 16"
        >
          <use
            xlinkHref="#burger"
          />
        </svg>
        alias 0
      </div>
      <svg
        className="css-e7nc70"
        viewBox="0 0 42 8"
      >
        <use
          xlinkHref="#flagSmall"
        />
      </svg>
      <div
        className="css-s5xdrg"
      >
        <svg
          className="css-1l8frzb"
          viewBox="0 0 14 14"
        >
          <use
            xlinkHref="#link"
          />
        </svg>
        <span
          className="css-1ezwu69"
        >
          uri 0
        </span>
      </div>
    </div>
  </div>
  <div
    className=""
  >
    <div>
      <div
        className="css-1ggp6x0"
      >
        <svg
          className="css-1f5lesb"
          viewBox="0 0 16 16"
        >
          <use
            xlinkHref="#burger"
          />
        </svg>
        alias 1
      </div>
      <svg
        className="css-e7nc70"
        viewBox="0 0 42 8"
      >
        <use
          xlinkHref="#flagSmall"
        />
      </svg>
      <div
        className="css-s5xdrg"
      >
        <svg
          className="css-1l8frzb"
          viewBox="0 0 14 14"
        >
          <use
            xlinkHref="#link"
          />
        </svg>
        <span
          className="css-1ezwu69"
        >
          uri 1
        </span>
      </div>
    </div>
  </div>
  <div
    className=""
  >
    <div>
      <div
        className="css-1ggp6x0"
      >
        <svg
          className="css-1f5lesb"
          viewBox="0 0 16 16"
        >
          <use
            xlinkHref="#burger"
          />
        </svg>
        alias 2
      </div>
      <svg
        className="css-e7nc70"
        viewBox="0 0 42 8"
      >
        <use
          xlinkHref="#flagSmall"
        />
      </svg>
      <div
        className="css-s5xdrg"
      >
        <svg
          className="css-1l8frzb"
          viewBox="0 0 14 14"
        >
          <use
            xlinkHref="#link"
          />
        </svg>
        <span
          className="css-1ezwu69"
        >
          uri 2
        </span>
      </div>
    </div>
  </div>
  <div
    className=""
  >
    <div>
      <div
        className="css-1ggp6x0"
      >
        <svg
          className="css-1f5lesb"
          viewBox="0 0 16 16"
        >
          <use
            xlinkHref="#burger"
          />
        </svg>
        alias 3
      </div>
      <svg
        className="css-e7nc70"
        viewBox="0 0 42 8"
      >
        <use
          xlinkHref="#flagSmall"
        />
      </svg>
      <div
        className="css-s5xdrg"
      >
        <svg
          className="css-1l8frzb"
          viewBox="0 0 14 14"
        >
          <use
            xlinkHref="#link"
          />
        </svg>
        <span
          className="css-1ezwu69"
        >
          uri 3
        </span>
      </div>
    </div>
  </div>
  <div
    className=""
  >
    <div>
      <div
        className="css-1ggp6x0"
      >
        <svg
          className="css-1f5lesb"
          viewBox="0 0 16 16"
        >
          <use
            xlinkHref="#burger"
          />
        </svg>
        alias 4
      </div>
      <svg
        className="css-e7nc70"
        viewBox="0 0 42 8"
      >
        <use
          xlinkHref="#flagSmall"
        />
      </svg>
      <div
        className="css-s5xdrg"
      >
        <svg
          className="css-1l8frzb"
          viewBox="0 0 14 14"
        >
          <use
            xlinkHref="#link"
          />
        </svg>
        <span
          className="css-1ezwu69"
        >
          uri 4
        </span>
      </div>
    </div>
  </div>
  <div
    className=""
  >
    <div>
      <div
        className="css-1ggp6x0"
      >
        <svg
          className="css-1f5lesb"
          viewBox="0 0 16 16"
        >
          <use
            xlinkHref="#burger"
          />
        </svg>
        alias 5
      </div>
      <svg
        className="css-e7nc70"
        viewBox="0 0 42 8"
      >
        <use
          xlinkHref="#flagSmall"
        />
      </svg>
      <div
        className="css-s5xdrg"
      >
        <svg
          className="css-1l8frzb"
          viewBox="0 0 14 14"
        >
          <use
            xlinkHref="#link"
          />
        </svg>
        <span
          className="css-1ezwu69"
        >
          uri 5
        </span>
      </div>
    </div>
  </div>
  <div
    className=""
  >
    <div>
      <div
        className="css-1ggp6x0"
      >
        <svg
          className="css-1f5lesb"
          viewBox="0 0 16 16"
        >
          <use
            xlinkHref="#burger"
          />
        </svg>
        alias 6
      </div>
      <svg
        className="css-e7nc70"
        viewBox="0 0 42 8"
      >
        <use
          xlinkHref="#flagSmall"
        />
      </svg>
      <div
        className="css-s5xdrg"
      >
        <svg
          className="css-1l8frzb"
          viewBox="0 0 14 14"
        >
          <use
            xlinkHref="#link"
          />
        </svg>
        <span
          className="css-1ezwu69"
        >
          uri 6
        </span>
      </div>
    </div>
  </div>
  <div
    className=""
  >
    <div>
      <div
        className="css-1ggp6x0"
      >
        <svg
          className="css-1f5lesb"
          viewBox="0 0 16 16"
        >
          <use
            xlinkHref="#burger"
          />
        </svg>
        alias 7
      </div>
      <svg
        className="css-e7nc70"
        viewBox="0 0 42 8"
      >
        <use
          xlinkHref="#flagSmall"
        />
      </svg>
      <div
        className="css-s5xdrg"
      >
        <svg
          className="css-1l8frzb"
          viewBox="0 0 14 14"
        >
          <use
            xlinkHref="#link"
          />
        </svg>
        <span
          className="css-1ezwu69"
        >
          uri 7
        </span>
      </div>
    </div>
  </div>
  <div
    className=""
  >
    <div>
      <div
        className="css-1ggp6x0"
      >
        <svg
          className="css-1f5lesb"
          viewBox="0 0 16 16"
        >
          <use
            xlinkHref="#burger"
          />
        </svg>
        alias 8
      </div>
      <svg
        className="css-e7nc70"
        viewBox="0 0 42 8"
      >
        <use
          xlinkHref="#flagSmall"
        />
      </svg>
      <div
        className="css-s5xdrg"
      >
        <svg
          className="css-1l8frzb"
          viewBox="0 0 14 14"
        >
          <use
            xlinkHref="#link"
          />
        </svg>
        <span
          className="css-1ezwu69"
        >
          uri 8
        </span>
      </div>
    </div>
  </div>
  <div
    className=""
  >
    <div>
      <div
        className="css-1ggp6x0"
      >
        <svg
          className="css-1f5lesb"
          viewBox="0 0 16 16"
        >
          <use
            xlinkHref="#burger"
          />
        </svg>
        alias 9
      </div>
      <svg
        className="css-e7nc70"
        viewBox="0 0 42 8"
      >
        <use
          xlinkHref="#flagSmall"
        />
      </svg>
      <div
        className="css-s5xdrg"
      >
        <svg
          className="css-1l8frzb"
          viewBox="0 0 14 14"
        >
          <use
            xlinkHref="#link"
          />
        </svg>
        <span
          className="css-1ezwu69"
        >
          uri 9
        </span>
      </div>
    </div>
  </div>
</div>
`;
